### ISSUE TITLE
[java] Unused Imports false positive for static import and Standard imports

### ISSUE THREAD

**Issue by vikhil1999** [pmd/pmd#4167]
Affects PMD Version: 
6.50.0
Rule: Unused Imports

https://pmd.github.io/pmd-6.4.0/pmd_rules_java_bestpractices.html#unusedimports

**Description:**
A false positive is generated for the Unused Imports rule on static import and Standard import, even if the methods of that import are being used. Using import static org.mockito.Mockito.*; instead of import static org.mockito.Mockito.mock, pmd shows unused import org.mockito.Mockito. The same issue applies to the standard import as well.

**Code Sample demonstrating the issue:**

```java
import com.some.thing.models.*; // This is shown as a violation [Incorrect]
import static org.mockito.Mockito.*; // This is shown as a violation [Incorrect]

public class Test {
private Service service = mock(Service.class);   // mock method comes from mockito import
private Mapper mapper = mock(Mapper.class);
 XYZ xyz;    // xyz is part of the models import
ABC abc;    // abc is part of the models import 

@Before
public void setup() throws Exception{
        xyz = JsonHelper.convertJsonFileToPojo("MockJson.json", XYZ.class);
        abc= JsonHelper.convertJsonFileToPojo("MockJson.json", ABC.class);
    }
}
```

**Expected outcome:**
PMD should not show violation for unused import as the methods of the import is being used in the code.
PMD reports a violation at line 1 and 2 but that's wrong. That's a false positive.

**Running PMD through:**
[ CLI ]

**Comment by jsotuyod** [comment]
@vikhil1999 thanks for the report. You don't clarify this, but I'm guessing you are not passing the [`--aux-classpath`](https://pmd.sourceforge.io/pmd-6.50.0/pmd_userdocs_cli_reference.html) to PMD.

With a proper auxclasspath PMD can effectively tell which class comes from where, and know if a star import is effectively unused or not. Without it PMD can't tell if `XYZ` is coming from package `com.some.thing.models` or is actually an automatic import from a class in the same package as the `Test` class. Without being able to discard it, PMD (as any static analysis tool) will err on the side of caution, flagging a *possible* violation (this is called *soundness*).

If you properly pass along your applications compile and runtime classpath (including compiled classes) to PMD through `--aux-classpath` the results will be significantly better. This is done automatically by most integrations (ie: Gradle, Maven, etc.) but has to be done manually through CLI.

**Comment by vikhil1999** [comment]
@jsotuyod Thanks for the quick response. Can you please give me an example on how to provide the --aux-classpath and what about the mockito will the same applies to mockito as well?

**Comment by jsotuyod** [comment]
As per the docs, there are 2 ways to set the auxclasspath:

1. Create a file with an entry per file. For instance, if the classes are being compiled to a "bin" dir, it could look like this (more entries would have to be added for each dependency):

auxclasspath.txt
```
libs/mockito-all-1.10.19.jar
bin/
```

and then just pass along `--aux-classpath file://path/to/auxclasspath.txt`

2. Alternatively, you can skip the `auxclasspath.txt` file and just pass the values formated as a classpath (under Windows, you would use `;` instead of `:`) as follows: `--aux-classpath "libs/mockito-all-1.10.19.jar:bin/"`
